### PR TITLE
Make windows_buildenv.bat work with spaces in path

### PR DESCRIPTION
**Pull request by fwolter**
If the Windows user name contains a space (and so the user directory), the execution of windows_buildenv.bat failed. This fixes it.

### CODE REVIEW

File: tools/windows_buildenv.bat [hunk 1]
```diff
@@ -150,7 +150,7 @@ EXIT /B 0
 
 :READ_ENVNAME
     ECHO Reading name of prebuild environment from "%MIXXX_ROOT%\packaging\windows\build_environment"
-    SET /P BUILDENV_NAME=<%MIXXX_ROOT%\packaging\windows\build_environment
+    SET /P BUILDENV_NAME=<"%MIXXX_ROOT%\packaging\windows\build_environment"
     SET BUILDENV_NAME=!BUILDENV_NAME:PLATFORM=%PLATFORM%!
     SET BUILDENV_NAME=!BUILDENV_NAME:CONFIGURATION=%CONFIGURATION%!
     SET RETVAL=%BUILDENV_NAME%
```
Comment (daschuer) [inline tools/windows_buildenv.bat:156]: Do we need quotes here as well?

Comment (fwolter) [reply tools/windows_buildenv.bat:156]: Yes, otherwise the batch file demands a user input.

**Comment by daschuer** [comment]
Thank you. CI is still green. However it would be nice to have a look from a windows user before merge. 

@JoergAtGithub
@qgazq

Can you confirm this is still working for you?

**Comment by daschuer** [comment]
Can this go to the 2.3 branch?

**Comment by JoergAtGithub** [comment]
I tried this, but it fails in a strange way:

Version from Main branch:
`C:\Users\Joerg\source\repos\JoergAtGithub\mixxx\tools>windows_buildenv.bat
Reading name of prebuild environment from "C:\Users\Joerg\source\repos\JoergAtGithub\mixxx\packaging\windows\build_environment"
Environment name: mixxx-deps-2.4-x64-windows-56c7437
Build environment path: C:\Users\Joerg\source\repos\JoergAtGithub\mixxx\buildenv\mixxx-deps-2.4-x64-windows-56c7437
Environment Variables:
- VCPKG_ROOT='C:\Users\Joerg\source\repos\JoergAtGithub\mixxx\buildenv\mixxx-deps-2.4-x64-windows-56c7437'
- CMAKE_GENERATOR='Ninja'
Generating "CMakeSettings.json"...
CMakeSettings.json already exists, creating backup at "CMakeSettings_2022-08-17_17-09-04.json"...
Drücken Sie eine beliebige Taste . . .`

Version from this PR:
`C:\Users\Joerg\source\repos\JoergAtGithub\mixxx\tools>windows_buildenv.bat
Reading name of prebuild environment from "C:\Users\Joerg\source\repos\JoergAtGithub\mixxx\packaging\windows\build_environment"
Environment name: mixxx-deps-2.4-x64-windows-56c7437
Build environment path: C:\Users\Joerg\source\repos\JoergAtGithub\mixxx\buildenv\mixxx-deps-2.4-x64-windows-56c7437
Environment Variables:
- VCPKG_ROOT='C:\Users\Joerg\source\repos\JoergAtGithub\mixxx\buildenv\mixxx-deps-2.4-x64-windows-56c7437'
- CMAKE_GENERATOR='Ninja'
Generating "CMakeSettings.json"...
CMakeSettings.json already exists, creating backup at "CMakeSettings_2022-08-17_17-09-53.json"...
Das Sprungziel - AddCMakeVar2CMakeSettings_JSON wurde nicht gefunden.
Syntaxfehler.`

The download and the unzipping (using 7zip) worked fine, but the CMakeSettings.json output is incomplete:
[CMakeSettings.json_Main.txt](https://github.com/mixxxdj/mixxx/files/9365146/CMakeSettings.json_Main.txt)
[CMakeSettings.json_PR4896.txt](https://github.com/mixxxdj/mixxx/files/9365148/CMakeSettings.json_PR4896.txt)

**Comment by fwolter** [comment]
That is really weird. The JSON file is generated until FFMPEG and then, out of a sudden, `AddCMakeVar2CMakeSettings_JSON` can't be found anymore? When I run this PR's batch file, it looks like this (Powershell and cmd.exe, with a space in the path and without)

```
C:\Users\Fabian Wolter\git\mixxx\tools>windows_buildenv.bat
Reading name of prebuild environment from "C:\Users\Fabian Wolter\git\mixxx\packaging\windows\build_environment"
Environment name: mixxx-deps-2.4-x64-windows-56c7437
Build environment path: C:\Users\Fabian Wolter\git\mixxx\buildenv\mixxx-deps-2.4-x64-windows-56c7437
Environment Variables:
- VCPKG_ROOT='C:\Users\Fabian Wolter\git\mixxx\buildenv\mixxx-deps-2.4-x64-windows-56c7437'
- CMAKE_GENERATOR='Ninja'
Generating "CMakeSettings.json"...
CMakeSettings.json already exists, creating backup at "CMakeSettings_2022-08-17_21-17-25.json"...
Drücken Sie eine beliebige Taste . . .
```

There's something fishy about the reported "Syntaxfehler". Is your windows_buildenv.bat corrupted maybe?

**Comment by JoergAtGithub** [comment]
I found the issue: My file had Unix line ends. I didn't checked out the whole PR, just downloaded this file. I guess this was my failure.
With correct line ends, it works as expected!

**Comment by daschuer** [comment]
Thank you for testing.

Did you consider to rebase this to the 2.3 branch? 
I will later merge 2.3 to main so it will land in main as well. Would this work or are the files too different?

**Comment by fwolter** [comment]
Tried, but there are some conflicts... I'd like to keep it on main in this PR.

**Review by daschuer — APPROVED** [review]
Thank you. LGTM

**Pull request merged**